### ISSUE TITLE
Nominatim update exception

### ISSUE THREAD

**Issue by andypl78** [komoot/photon#229]
This is normal ?

Debian Jessie

```

java version "1.7.0_101"
OpenJDK Runtime Environment (IcedTea 2.6.6) (7u101-2.6.6-2~deb8u1)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)

```

 java -jar -Xms2048m -Xmx10240m photon-0.2.7.jar -host localhost -port 5432 -database nominatim -user nominatim -password ... -languages en

```
[elasticsearch[Midgard Serpent][clusterService#updateTask][T#1]] INFO org.elasticsearch.gateway - [Midgard Serpent] recovered [1] indices into cluster_state
== Spark has ignited ...
>> Listening on 0.0.0.0:2322
[Thread-1] INFO org.eclipse.jetty.server.Server - jetty-9.0.z-SNAPSHOT
[Thread-1] INFO org.eclipse.jetty.server.ServerConnector - Started ServerConnector@1a9bd9bb{HTTP/1.1}{0.0.0.0:2322}
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 1
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 2
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 3
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 4
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 5
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 6
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 7
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 8
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 9
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 10
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 11
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 12
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 13
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 14
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 15
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 16
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 17
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 18
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 19
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 20
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 21
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 22
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 23
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 24
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 25
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 26
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 27
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 28
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 29
[Thread-44] INFO de.komoot.photon.nominatim.NominatimUpdater - Starting rank 30
Exception in thread "Thread-44" org.elasticsearch.action.ActionRequestValidationException: Validation Failed: 1: no requests added;
        at org.elasticsearch.action.ValidateActions.addValidationError(ValidateActions.java:29)
        at org.elasticsearch.action.bulk.BulkRequest.validate(BulkRequest.java:476)
        at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:67)
        at org.elasticsearch.client.node.NodeClient.execute(NodeClient.java:98)
        at org.elasticsearch.client.support.AbstractClient.bulk(AbstractClient.java:167)
        at org.elasticsearch.action.bulk.BulkRequestBuilder.doExecute(BulkRequestBuilder.java:166)
        at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:91)
        at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:65)
        at de.komoot.photon.elasticsearch.Updater.updateDocuments(Updater.java:64)
        at de.komoot.photon.elasticsearch.Updater.finish(Updater.java:30)
        at de.komoot.photon.nominatim.NominatimUpdater.update(NominatimUpdater.java:65)
        at de.komoot.photon.App$1$1.run(App.java:159)
```

**Comment by christophlingg** [comment]
is your database empty?

**Comment by nawaz2013** [comment]
Hi,

I am witnessing the same issue while trying to update. I have updated a recored in  'placex' table manually. I would like to make an update without importing from scratch since it takes around an hour in my settings to rebuild photon_data from scratch.

Please guide if this is possible.

**Comment by MJulaie** [comment]
I have the same issue for updating nominatim.Everything is work so I guess that bugs riase from this little problem in es Updater: 
https://stackoverflow.com/questions/22996337/elasticsearch-no-requests-added-bulk-api-error

**Comment by amirmasud** [comment]
Hi.
It's an issue if you  access  /nominatim-update while no changes are available in nominatim database with index_status != 0
Your problem will happen if you run nominatim index by yourself with `utils/update.php --index ` command before accessing this url.
Photon takes care of indexing nominatim and works with indexed_status column in nominatim database. So if you run aforementioned command it will unset the indexed_status of the whole changes and Photon won't notified about changes.Thus it thinks no changes are available and raises that exception. 

@christophlingg It's better not to call finish function whenever nothing is available to update to prevent the exception.

**Comment by pbrugger** [comment]
So the correct way of updating would be like this?
1) import osm changeset to nominatim e.g. 
`./utils/update.php --import-diff filename.osc`
2) run the update with e.g.
`http://localhost:2323/nominatim-update`
3) all new and changed data should be available via local nominatim search and photon search
Am I right?

I accidentally did the `./utils/update.php --index` after step 1  and so I got the `org.elasticsearch.action.ActionRequestValidationException`

**Comment by amirmasud** [comment]
Yes. All new and changed data should be available via local nominatim search and photon search 
(nominatim will be updated only If you set  defaultAutoCommit of datasource to true. or do some other transaction managements. It's an issue I addressed in issue #257 )

Whenever there is no doc is available to apply to elasticsearch ActionRequestValidationException error occurs. 

(So if you call `/nominatim-update`  when no changes are available this exception will be thrown too)

**Comment by RhinoDevel** [comment]
Maybe [this change](https://github.com/komoot/photon/pull/382) fixes the problem?

See 

> remove code that tried to set indexed_status to 0 in the source
> Nominatim database (running update.php --index will do that)

In its description.

**Comment by lonvia** [comment]
Yes, updates are fixed with #382.
